packages/frontend/components/auth/verification/resend-code/component.tsx

```diff
@@ -0,0 +1,52 @@
+import { useState, useEffect } from 'react';
+import { useTranslation } from 'next-i18next';
+import { resendCodeBlock, resendCodeBtn, resendCodeTxt } from './styles';
+import type { ResendCodeButtonProps } from './types';
+
+export const ResendCodeButton = ({
+  onClickResend,
+  timeLimit,
+}: ResendCodeButtonProps) => {
+  const { t } = useTranslation('auth');
+  const { t: commonLang } = useTranslation('common');
+
+  const [resendDisabled, setResendDisabled] = useState(false);
+  const [timeToResend, setTimeToResend] = useState(timeLimit);
+
+  useEffect(() => {
+    if (resendDisabled) {
+      const interval = setInterval(() => {
+        setTimeToResend((prev) => prev - 1);
+      }, 1000);
```

Comment:
We are using this timeout in several places, maybe it make sense to have it constants for